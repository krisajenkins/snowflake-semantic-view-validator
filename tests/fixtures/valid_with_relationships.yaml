name: WITH_RELATIONSHIPS
description: This semantic model includes relationships and custom queries
tables:
  - name: CUSTOMERS
    base_table:
      database: SALES_DB
      schema: PUBLIC
      table: CUSTOMERS
    dimensions:
      - name: CUSTOMER_ID
        expr: CUSTOMER_ID
        data_type: VARCHAR
        unique: true
      - name: CUSTOMER_NAME
        expr: CUSTOMER_NAME
        data_type: VARCHAR
  - name: ORDERS
    base_table:
      database: SALES_DB
      schema: PUBLIC
      table: ORDERS
    dimensions:
      - name: ORDER_ID
        expr: ORDER_ID
        data_type: VARCHAR
      - name: CUSTOMER_ID
        expr: CUSTOMER_ID
        data_type: VARCHAR
    time_dimensions:
      - name: ORDER_DATE
        expr: ORDER_DATE
        data_type: DATE
    facts:
      - name: AMOUNT
        expr: AMOUNT
        data_type: FLOAT
        aggregation: sum
relationships:
  - name: customer_orders
    left_table: CUSTOMERS
    right_table: ORDERS
    join_type: LEFT
    join_condition: CUSTOMERS.CUSTOMER_ID = ORDERS.CUSTOMER_ID
custom_queries:
  - name: top_customers
    description: Find the top 10 customers by order amount
    sql: SELECT CUSTOMER_NAME, SUM(AMOUNT) as TOTAL FROM CUSTOMERS JOIN ORDERS USING (CUSTOMER_ID) GROUP BY CUSTOMER_NAME ORDER BY TOTAL DESC LIMIT 10
