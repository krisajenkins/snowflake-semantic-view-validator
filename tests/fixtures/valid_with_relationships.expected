════════════════════════════════════════════════════════════════════════════════
  SEMANTIC MODEL VALIDATION SUMMARY
════════════════════════════════════════════════════════════════════════════════

Name: WITH_RELATIONSHIPS
Description: This semantic model includes relationships and custom queries

TABLES
────────────────────────────────────────────────────────────────────────────────
  • CUSTOMERS
    Location: SALES_DB.PUBLIC.CUSTOMERS
    Dimensions: 2
    Time Dimensions: 0
    Facts: 0

  • ORDERS
    Location: SALES_DB.PUBLIC.ORDERS
    Dimensions: 2
    Time Dimensions: 1
    Facts: 1

RELATIONSHIPS
────────────────────────────────────────────────────────────────────────────────
  • customer_orders
    LEFT CUSTOMERS → ORDERS

CUSTOM QUERIES
────────────────────────────────────────────────────────────────────────────────
  • top_customers
    Find the top 10 customers by order amount

════════════════════════════════════════════════════════════════════════════════
✓ Validation successful!
════════════════════════════════════════════════════════════════════════════════